File: technomarket.entity/Category.cs
using System;
using System.Collections.Generic;

namespace technomarket.entity
{
    public class Category
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public ICollection<SubCategory> SubCategories { get; set; } = new List<SubCategory>();
        public ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: technomarket.webui/src/app/models/category.ts
import { ProductPhoto } from "./productPhoto";

export interface Category {
    id: string,
    name: string,
    subCategories?: SubCategory[]

}

export interface SubCategory {
    id: string,
    name: string,
    products: ProductBasicModel[]
}

export interface ProductBasicModel {
    id: string,
    name: string,
    price: number,
    quantity: number,
    isApproved: boolean,
    isHome: boolean,
    description: string,
    photos: ProductPhoto[]
}